note-new

```diff
@@ -1,15 +1,47 @@
 #!/bin/bash
 
-required_date="${1:-today}"
+# Used to check which machine we are running on
+unameOut="$(uname -s)"
+MAC_OS="Darwin"
+LINUX_OS="Linux"
+
+# Use date from args, or default, take into account date implementation differences
+if [ "${unameOut}" = "${LINUX_OS}" ]
+then
+  requiredDate="${1:-today}"
+elif [ "${unameOut}" = "${MAC_OS}" ]
+then
+  # On BSD based systems (i.e. Mac), date cannot accept any input as a string
+  # So we have to enforce a specific format to validate against
+  macOsDefaultDateFormat="%Y-%m-%d"
+  todaysDateMacOs="$(date -j +${macOsDefaultDateFormat})"
+  requiredDate="${1:-todaysDateMacOs}"
+fi
 
 rootFolder="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 rootFolder="$( cat "${rootFolder}/.noteFolderPath")"
+
 if [ -z $rootFolder ]; then
   echo "Can't find root folder '${rootFolder}', did you run note-setup?"
   exit 1
 fi
 
-formated_date="$( date -d "${required_date}" +'%Y-%m-%d' )"
+# Validate input date and format it
+# Take into account different implementations of `date` commant
+elif [ "${unameOut}" = "${LINUX_OS}" ]
```

Comment:
should be `if`?

Reply:
Must of been a typo ! Seems gone now. WIll check other files in case I copied and pasted this somewhere